### PR TITLE
Net tests switch to new net_pkt/net_context APIs

### PR DESCRIPTION
**Pull request by tbursztyka**
Switching tests to the new APIs so #13618 will be merged

### CODE REVIEW

**Review by rveerama1 — APPROVED** [review]

**Comment by pfalcon** [comment]
> +813 −1,993

So, this is again, not just "switch", but also removal of more than 1000 lines of tests.

**Comment by tbursztyka** [comment]
@pfalcon take 30 seconds next time to see the diffs and the commit messages before saying this.

File: tests/net/net_pkt/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 cmake_minimum_required(VERSION 3.13.1)
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
-project(net_pkt)
+project(NONE)
```
Comment (pfalcon) [inline tests/net/net_pkt/CMakeLists.txt]: This isn't change for good, is it?

File: tests/net/net_pkt/src/main.c [hunk 2]
```diff
@@ -1,874 +1,617 @@
...
-
-static const char example_data[] =
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?";
-
-static void test_ipv6_multi_frags(void)
```
Comment (pfalcon) [inline tests/net/net_pkt/src/main.c]: Well, this commit raises concerns that some previous testcases might be lost.

**Comment by pfalcon** [comment]
@tbursztyka 

> @pfalcon take 30 seconds next time to see the diffs and the commit messages before saying this.

Well, I exactly took some 30 seconds for the overview of the PR, and that's what I saw. And given the case of one of the previous PRs (sorry, don't have a link handy), I had some concerns.

More detailed review took much, much more time, especially including  https://github.com/zephyrproject-rtos/zephyr/issues/14093 & https://github.com/zephyrproject-rtos/zephyr/pull/14095 . You actually could simplify review process for fellow reviewers by adding suitable summary info for the PR, e.g. "Noticeable reduction in line count is due to the fact we use net_context_send_new() which replaces a number of calls in the older API".

I still have a concern that the last commit, " tests/net: Replace the legacy net_pkt API test by the new one." might lose some useful tests. Perhaps, you could explain why it doesn't?

**Review by jukkar — APPROVED** [review]

**Pull request merged**